src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java

```diff
@@ -328,8 +328,8 @@ private static Map<String, String> getMessages(ModuleDetails moduleDetails) {
      * @return a version string based on the package implementation version
      */
     private static String replaceVersionString(String report) {
-        final String version = SarifLogger.class.getPackage().getImplementationVersion();
-        return report.replace(VERSION_PLACEHOLDER, String.valueOf(version));
```

Comment:
String.valueOf(String) makes no sense this is an bug in the framework as the change is unrelated but enforced.